File: Next with CosmosDB/server/Basketball/Microservice/Models/ValueObjects/TeamScalation.cs
using NBA.Infrastructure;

namespace NBA.Models
{
    public class TeamScalation : BasketballValueObject
    {
        public string Id { get; set; }

        public string TeamName { get; set; }

        public List<PlayerSelection> Players { get; set; } = [];

        public override (bool Success, string Message) Validate()
        {
            //To be improved with FluentValidation library
            if (Id is null)
                return (false, "id cannot be null");

            if (TeamName is null)
                return (false, "teamName cannot be null");

            if (Players is null)
                return (false, "players cannot be null");

            foreach (var player in Players)
            {
                var (Success, Message) = player.Validate();
                if (!Success)
                    return (false, $"player of id '{player.PlayerId}' is invalid due to '{Message}'");
            }

            return (true, null);
        }
    }
}

File: React-Next with DotNet-CosmosDB/client/nba-statistics/src/models/TeamScalation.ts
import { PlayerSelection } from "./PlayerSelection";

export class TeamScalation {
    teamId: string;
    teamName: string;
    players: PlayerSelection[]

    constructor(teamId: string, teamName: string, players: PlayerSelection[]) {
        this.teamId = teamId;
        this.teamName = teamName;
        this.players = players;
    }
}